### PR TITLE
[NUI] Add GLWindow Rendering mode

### PR DESCRIPTION
**Pull request by sparrow74**
Add GLWindow rendering mode.
It has two mode as continuous and on demand.
If the mode is continuous, the rendering user callback is called continuously.
If the mode is on demand, the rendering user callback is called by call renderOnce().

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/Window/GLWindow.cs [hunk 1]
```diff
@@ -494,6 +494,50 @@ public void Destroy()
...
 
+        /// <summary>
+        /// Enumeration for rendering mode
+        /// This Enumeration is used to choose the rendering mode.
+        /// It has two options.
+        /// One of them is continuous mode. It is rendered continuously.
+        /// The other is on demand mode. It is rendered by application.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public enum GLWindowRenderingMode
```
Comment (huiyueun) [inline src/Tizen.NUI/src/public/Window/GLWindow.cs:505]: How about changing 'RenderingModeType'?

Comment (sparrow74) [reply src/Tizen.NUI/src/public/Window/GLWindow.cs:505]: Thank you for your kind review.
Actually, DALi has the similar enumeration. The name is "RenderingMode".
I think same name is better.

**Review by dongsug-song — COMMENTED** [review]
Hello, 
could you check the warning in line 149, 155?

**Remove the underscores from member name Tizen.NUI.GLWindow.GLESVersion.Version_2_0.**

could you change them as PascalCase?

**Comment by dkdk-ryu** [comment]
In OpenTK case, They name the namespaces as OpenTK.Graphics.ES20 and ES30.
https://docs.microsoft.com/en-us/dotnet/api/opentk.graphics.es20?view=xamarin-ios-sdk-12
I think It should be better to follow the naming of OpenTK and don't break the Tizen rule.

**Comment by sparrow74** [comment]
Thank you for detail review.
NUI has the enumeration and OpenTK has the namespace.
I think it is different case.
In addition, the Enumeration's name is "GLESVerion".
So, The prefix's candidates are 'GLES", "Version", "Ver" or "V".
If "ES" is the best choice, the enumeration's name should be changed with "ESVersion".
Could you share your opinion?

**Comment by dkdk-ryu** [comment]
OK, I think Version20 is better.

**Comment by sparrow74** [comment]
I'm appreciated your review.
I will update with your opinion.

**Comment by dkdk-ryu** [comment]
I think GLView would need GLRenderingMode.
So how about moving GLRenderingMode to NUIConstants.cs?

**Comment by sparrow74** [comment]
@dkdk-ryu  
You're right.
GLRenderingMode should be moved to NUIConstants.cs for GLView.
In addition, GLESVersion enum should be moved, too.
Thank you for your review. :)

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Comment by dkdk-ryu** [comment]
Looks good to me.

**Pull request merged**